src/shared/identityUtils.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+import { CryptoUtils } from '@salesforce/lwc-dev-mobile-core';
+import { LightningDevConfig } from './lightningDevConfig.js';
+import { ConfigVars } from './../configMeta.js';
+
+export class IdentityUtils {
+  public static async updateConfigWithIdentityToken(): Promise<void> {
+    const token = await this.getIdentityToken();
+    if (!token) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const generatedIdentityToken = CryptoUtils.generateIdentityToken();
+      await this.writeIdentityToken(generatedIdentityToken);
+    }
+    return Promise.resolve();
+  }
+
+  public static async getIdentityToken(): Promise<string | undefined> {
+    const config = await LightningDevConfig.create({
+      isGlobal: false,
+    });
+    await config.read();
```

Comment:
I would update the method signature to take an optional `config` param so that we don't have to re-read the config if it was already read before. Then you can check to see if `config` param was passed in or not and if not only then proceed with  these lines. This way we can reduce disk access/read